docs/src/faq.md

```diff
@@ -0,0 +1,402 @@
+    end
+    @inbounds for j in axes(B, 2), i in axes(A, 1)
+        @inbounds @simd for k in axes(A,2)
+            R[i,j] += A[i,k] * B[k,j]
+        end
+    end
+    nothing
+end
+
+Random.seed!(1234)
+A = rand(5, 3)
+B = rand(3, 7)
+
+R = zeros(size(A,1), size(B,2))
+∂z_∂R = rand(size(R)...)  # Some gradient/tangent passed to us
+∂z_∂R0 = copyto!(similar(∂z_∂R), ∂z_∂R)  # exact copy for comparison
+
+∂z_∂A = zero(A)
+∂z_∂B = zero(B)
+
+Enzyme.autodiff(Reverse, mymul!, Const, Duplicated(R, ∂z_∂R), Duplicated(A, ∂z_∂A), Duplicated(B, ∂z_∂B))
+```
+
+Now we have:
+
+```@example pullback
+R ≈ A * B            &&
+∂z_∂A ≈ ∂z_∂R0 * B'   &&  # equivalent to Zygote.pullback(*, A, B)[2](∂z_∂R)[1]
+∂z_∂B ≈ A' * ∂z_∂R0       # equivalent to Zygote.pullback(*, A, B)[2](∂z_∂R)[2]
+```
+
+Note that the result of the backpropagation is *added to* `∂z_∂A` and `∂z_∂B`, they act as accumulators for gradient information.
+
+## Identical types in `Duplicated`
+
+Enzyme forces `x` and `∂f_∂x` to have the same types when constructing objects of type `Duplicated`, `DuplicatedNoNeed`, `BatchDuplicated`, etc.
+This is not a mathematical requirement, but rather a guardrail to prevent user error.
+The memory locations of `x` and `∂f_∂x` are accessed in the same way by the differentiation code, so they should have the same data layout.
+Equality of types is an approximation of this condition.
+
+In some ways, type equality is too strict: two different types can have the same data layout.
+But in other ways it is not strict enough.
+For instance, if `x` and `∂f_∂x` are sparse arrays, their sparsity pattern should be identical, but this is not encoded in the type.
+To make sure that `∂f_∂x` has the right data layout, create it with `∂f_∂x = Enzyme.make_zero(x)`.
+
+## CUDA support
+
+[CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) is only supported on Julia v1.7.0 and onwards. On v1.6, attempting to differentiate CUDA kernel functions will not use device overloads
+correctly and thus returns fundamentally wrong results.
+
+## Sparse arrays
+
+At the moment there is limited support for sparse linear algebra operations. Sparse arrays may be used, but care must be taken because backing arrays drop zeros in Julia (unless told not to).
+
+```jldoctest sparse
+using SparseArrays
+a = sparse([2.0])
+da1 = sparse([0.0]) # Incorrect: SparseMatrixCSC drops explicit zeros
+Enzyme.autodiff(Reverse, sum, Active, Duplicated(a, da1))
+da1
+
+# output
+
+1-element SparseVector{Float64, Int64} with 0 stored entries
+```
+
+```jldoctest sparse
+da2 = sparsevec([1], [0.0]) # Correct: Prevent SparseMatrixCSC from dropping zeros
+Enzyme.autodiff(Reverse, sum, Active, Duplicated(a, da2))
+da2
+
+# output
+
+1-element SparseVector{Float64, Int64} with 1 stored entry:
+  [1]  =  1.0
+```
+
+Sometimes, determining how to perform this zeroing can be complicated.
+That is why Enzyme provides a helper function `Enzyme.make_zero` that does this automatically.
+
+## Activity analysis
```

Comment:
This is much more specific than just activity analysis, so probably keeping the old name is more apt, or alternatively adding much more to describe what activity analysis is.

Reply:
I was trying to find a name that fits nicely in the TOC on the left side, that's why I shortened it. And it is also the first time in the docs that activity analysis is mentioned at all

Reply:
But I can roll it back if necessary

Reply:
renamed to "Activity of temporary storage" and fixed the link